main_test.go

```diff
@@ -56,32 +56,87 @@ func BenchmarkKubeStateMetrics(t *testing.B) {
 	builder.WithKubeClient(kubeClient)
 	builder.WithNamespaces(options.DefaultNamespaces)
 
-	collectors := builder.Build()
+	// This test is not suitable to be compared in terms of time, as it includes
```

Comment:
Maybe you can just pause the timer while you wait for the cache to be filled, by using `b.StopTimer()` right before `time.Sleep` is called and after that call `b.StartTimer`.

Reply:
I do agree with @LiliC, there are two benchmarks that we want to measure here:

1) filling the cache
2) rendering the output from the cache

It could be interesting to have an additional one that does do both (like we have now) just for overall performance, but in order to see gain of optimizations for each of the two scenarios above it would be good to have the detailed ones as well. They're distinct problems, that need distinct optimizations, so they should have distinct benchmarks.

Reply:
Not sure I understand. Filling cache and rendering into http response writer are split up into two subtests, thereby their results are reported as two distinct tests. Would you prefer to complete separate once @brancz?

Reply:
Ah that makes perfect sense. If that automatically resets the timers and bytes processed then this lgtm